scripts/planktoscope/segmenter/__init__.py

```diff
@@ -813,11 +813,16 @@ def segment_path(self, path, ecotaxa_export):
             )
         )
 
-        project = self.__global_metadata["sample_project"].replace(" ", "_")
-        sample = self.__global_metadata["sample_id"].replace(" ", "_")
-        date = datetime.datetime.utcnow().isoformat()
+        def clean_name(name):
+            s1="".join(c for c in name if c.isalnum() or c == " ")
+            s1 = s1.replace(" ","_")
+            return s1
 
-        self.__global_metadata["process_datetime"] = date
+        project = clean_name(self.__global_metadata["sample_project"])
+        sample = clean_name(self.__global_metadata["sample_id"])
+        date = clean_name(self.__global_metadata["object_date"])
```

Comment:
Nice catch, but shouldn't the date in the filename reflect when it was generated instead of the sample date?

Reply:
the problem is the aquisition date is always the same (14.09.2019 I think) so no way of distinguishing samples
How could the Raspberry get the real time/date?